src/middleware.ts

```diff
@@ -116,3 +116,48 @@ export const combine = <
       api as StoreApi<PrimaryState>
     )
   )
+
+export type StateStorage = {
+  getItem: (key: string) => string | null | Promise<string | null>
+  setItem: (key: string, value: string) => void | Promise<void>
```

Comment:
```suggestion
  getItem: (name: string) => string | null | Promise<string | null>
  setItem: (name: string, value: string) => void | Promise<void>
```

Just minor, but this should be more consistent?